### PR TITLE
Caiman modifications to handle new image types

### PR DESCRIPTION
**Pull request by pyhalov**

### CODE REVIEW

File: usr/src/cmd/slim-install/svc/media-fs-root [hunk 1]
```diff
@@ -285,9 +285,15 @@ elif [ $LIVECD = 1 ]; then
 	livemode=`$PRTCONF -v|$SED -n '/livemode/{;n;p;}'|$CUT -f 2 -d\'`
 	if [ "$livemode" = "text" ]; then
 		/usr/sbin/svcadm disable -t application/graphical-login/gdm
+		/usr/sbin/svcadm disable -t application/graphical-login/lightdm
 	else
 		[ "$livemode" = "vesa" ] && cp /.cdrom/jack/xorg.conf.vesa /etc/X11/xorg.conf
-		/usr/sbin/svcadm enable application/graphical-login/gdm
+		if [ -f /usr/sbin/lightdm ]; then
+			[ -f /usr/sbin/gdm ] && /usr/sbin/svcadm disable application/graphical-login/gdm
```
Comment (xen0l) [inline usr/src/cmd/slim-install/svc/media-fs-root:292]: Is GDM enabled  by default? Can't we disable it?

File: usr/src/cmd/distro_const/slim_cd/slimcd_boot_archive_configure [hunk 2]
```diff
@@ -85,21 +85,32 @@ SED=/usr/bin/sed
...
+	# It's OK that symbolic links have been created to these files already, as
+	# the files can change without affecting the links to them.
+
+	# Enable auto-login in gdm, and save original to replace after installation
+	$MKDIR -p ${PKG_IMG_PATH}/save/etc/gdm
+	$CP -p ${BA_BUILD}/etc/gdm/custom.conf ${PKG_IMG_PATH}/save/etc/gdm
+	$NAWK '/^\[daemon\]/ { print $0;
+	    print "AutomaticLoginEnable=true";
+	    print "AutomaticLogin=jack";
+	    print "GdmXserverTimeout=30";
```
Comment (xen0l) [inline usr/src/cmd/distro_const/slim_cd/slimcd_boot_archive_configure:99]: What about such settings for LightDM? Do they exist?

File: usr/src/cmd/distro_const/slim_cd/slimcd_boot_archive_configure [hunk 3]
```diff
@@ -85,21 +85,32 @@ SED=/usr/bin/sed
...
+	    continue }  { print $0 }' ${BA_BUILD}/etc/gdm/custom.conf \
+	    > ${TMP_DIR}/custom.conf
+	$CP ${TMP_DIR}/custom.conf ${BA_BUILD}/etc/gdm
```
Comment (xen0l) [inline usr/src/cmd/distro_const/slim_cd/slimcd_boot_archive_configure:102]: Why not redirecting directly to ${BA_BUILD}/etc/gdm and remove cp/rm step?

File: usr/src/lib/libict_pymod/ict.py [hunk 4]
```diff
@@ -1018,8 +1020,13 @@ def generate_sc_profile(self):
             return 0
 
         info_msg('Detected ' + self.keyboard_layout + ' keyboard layout')
-        status = _cmd_status('/usr/bin/sed s/US-English/' + \
+
+        if os.path.isfile(self.basedir + '/usr/sbin/lightdm'):
+            display_manager='lightdm' 
+            
+        status = _cmd_status('/usr/bin/sed -e s/US-English/' + \
```
Comment (xen0l) [inline usr/src/lib/libict_pymod/ict.py:1027]: Why is this needed? Comment would help. I suppose this is related to gdm pieces being in illumos-gate?

File: usr/src/cmd/distro_const/slim_cd/slimcd_boot_archive_configure (hunk 3 above)
Comment (pyhalov) [inline usr/src/cmd/distro_const/slim_cd/slimcd_boot_archive_configure:102]: sorry, another reason, it'll be `<custom.conf >custom.conf` pipeline

File: usr/src/lib/libict_pymod/ict.py [hunk 5]
```diff
@@ -2517,19 +2524,20 @@ def setup_sudo(self, login):
...
-                    if not l.startswith("jack"):
-                        fp_tmp.write(l)
-                if login:
-                    fp_tmp.write(login + ' ALL=(ALL) ALL\n')
-
-            os.remove(self.sudoers)
-            shutil.move(temp_file, self.sudoers)
-            os.chmod(self.sudoers, S_IREAD | S_IRGRP)
-            os.chown(self.sudoers, 0, 0) # chown root:root
+            if os.path.isfile(self.sudoers):
```
Comment (xen0l) [inline usr/src/lib/libict_pymod/ict.py:2527]: Generally, in Python it is a practice to not do things like if os.path.isfile, but try the action directly, e.g. open. It's considered to be more Pythonic. It is also considered an anti-pattern: http://docs.quantifiedcode.com/python-anti-patterns/readability/asking_for_permission_instead_of_forgiveness_when_working_with_files.html

File: usr/src/cmd/slim-install/svc/media-fs-root (hunk 1 above)
Comment (pyhalov) [inline usr/src/cmd/slim-install/svc/media-fs-root:292]: It should be enabled by default, if exists, as we do it in profile for live image.

File: usr/src/cmd/distro_const/slim_cd/slimcd_boot_archive_configure (hunk 3 above)
Comment (xen0l) [inline usr/src/cmd/distro_const/slim_cd/slimcd_boot_archive_configure:102]: Ah, right. Looks good.

File: usr/src/lib/libict_pymod/ict.py (hunk 4 above)
Comment (pyhalov) [inline usr/src/lib/libict_pymod/ict.py:1027]: slim_source already delivers one custom sc profile. Now we should just add one action for enabling gdm or lightdm. If we do it this way, it doesn't matter where these pieces are. Perhaps, it even would be better if they were removed from illumos-gate.

File: usr/src/cmd/distro_const/slim_cd/slimcd_boot_archive_configure (hunk 2 above)
Comment (pyhalov) [inline usr/src/cmd/distro_const/slim_cd/slimcd_boot_archive_configure:99]: Don't see one
